odoo/addons/base/models/res_lang.py

```diff
@@ -332,6 +332,26 @@ def write(self, vals):
             self.env['ir.default'].discard_values('res.partner', 'lang', lang_codes)
 
         res = super(Lang, self).write(vals)
+
+        # If we activate a lang, set it's url_code to the shortest version if
+        # possible
+        if vals.get('active'):
```

Comment:
Do we need to set the short form when the language holding it is disabled :thinking:?
E.g., if we have `"fr_BE"` (with `url_code == 'fr'`) and `"fr_CA"` (`url_code == 'fr_CA'`) installed, do we need to set `"fr"` on `"fr_CA"` when disabling `"fr_BE"` ?

Reply:
Doesn't cost too much to support it, I added some code to do it, and covered it in the tests.
With that change, now the behavior is complete, whatever the flow, if a lang can be shortened it will be done.